Code-4-Community/speak-for-the-trees-frontend
# Redirect After Login

## Why

[Monday.com Ticket](https://code4community-team.monday.com/boards/925293934/pulses/1016837535)


Redirects the user to the home page after they log in/sign up. Creates a `ROUTE` enum to make altering routes easier.

## This PR


- Get privilege level to check whether the user is logged in, if they are, redirect to home
- If the login/signup request fails, displays an error message
- Edited login page and form to list all window types in the switch case
- Login form checks whether the input in the email field follows the email pattern
- Styling changes to make tablet views neater (no overflow on signup form)

## Screenshots

 
<img width="1000" alt="LoginError" src="https://user-images.githubusercontent.com/22990100/107174875-5a402c80-6999-11eb-87b8-ea154b4a21e9.png">
<img width="1000" alt="SignupError" src="https://user-images.githubusercontent.com/22990100/108310862-1b248f00-7182-11eb-842e-af227db4977e.png">

## Verification Steps


Ran frontend and backend locally, checked all window types:
- entered incorrect login to see if error displayed correctly
- entered correct login for a user with standard privilege level to see redirect to home
- entered correct login for a user with admin privilege level to see redirect to home 
- entered signup request with an invalid username to see if error displayed correctly
- entered signup request with an invalid email to see if error displayed correctly
- entered valid signup request to see redirect to home
